testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,165 @@
+import "src/proto/grpc/testing/control.proto";
+
+service TestSessions {
+  // Starts a set of test scenarios.
+  //
+  // This is a long-running operation that is managed by the Operations service
+  // on the same server.  The long-running operation will assign a unique
+  // identifier to the session.
+  //
+  // The unique identifier can be used to poll for the session's status and
+  // results.  Streaming is not supported at present.
+  //
+  // It can also be used to cancel while in progress, but cancellations operate
+  // like killing a running process. If they occur while results are being
+  // reported, the persistent storage may receive some but not all of them.
+  //
+  // For the specification of a google.longrunning.Operation message, see the
+  // Long-running AIP at https://aip.dev/151.
+  rpc StartTestSession(StartTestSessionRequest) returns (google.longrunning.Operation) {
+    option (google.longrunning.operation_info) = {
+      response_type: "TestSessionResults"
+      metadata_type: "TestSessionMetadata"
+    };
+  }
+}
+
+message Container {
+  enum Kind {
+    // No kind was specified, so the service will attempt to infer one.
+    //
+    // This inference works as follows:
+    //   1. If this field belongs to the driver_container of the
+    //      StartTestSessionRequest, this kind is inferred to be a driver.
+    //   2. Otherwise, this field is inferred to be a client worker.
+    UNSPECIFIED = 0;
+
+    // Orchestrates the test amongst workers and reports results.
+    DRIVER = 1;
+
+    // Processes incoming requests from client containers.
+    SERVER = 2;
+
+    // Sends outgoing requests to a server container.
+    CLIENT = 3;
+  }
+
+  // The name of the image as it appears in the registry.
+  //
+  // When using GCR, this will likely be a fully-qualified URL. For example,
+  // the name of the Java worker may be "gcr.io/grpc-testing/java_qps_worker".
+  string name = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The kind which assigns the responsibilities of this container.
+  Kind kind = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  oneof version_identifier {
+    // The hash assigned to the image.
+    string sha256 = 2;
+
+    // A tag which was explicitly set on the image.
+    //
+    // For example, a developer may explicitly tag an image for repetitive
+    // testing and sharing. They may add a tag with a PR number or version.
+    //
+    // Note: the "latest" tag is automatically set by the registry, and there
+    // are no guarantees that it will point to a specific image. It should be
+    // avoided.
+    string tag = 3;
+  }
+
+  // The number of pods that should be created with this container. For example,
+  // specify 2 replicas on a client container to create 2 clients.
+  //
+  // This field will always be ignored on driver containers. It is ignored on
+  // server containers at present, but may be supported eventually.
+  int32 replicas = 4;
```

Comment:
Replicas allows a developer to create multiple versions of the same container while avoiding redundancy. It makes the **`StartTestSessionRequest`** message minimal for the most common use case of 2 clients. In the future, it could possibly be extended to multiple servers. We could use a kubernetes "service" and load balance to them. This likely replicates the environment that many customers of gRPC have.